packages/wrangler/src/api/pages/deploy.tsx

```diff
@@ -139,15 +148,35 @@ export async function deploy({
 		isProduction = project.production_branch === branch;
 	}
 
-	const deploymentConfig =
-		project.deployment_configs[isProduction ? "production" : "preview"];
-	const nodejsCompat =
-		deploymentConfig.compatibility_flags?.includes("nodejs_compat") ?? false;
+	const env = isProduction ? "production" : "preview";
+	const deploymentConfig = project.deployment_configs[env];
+	let config: Config | undefined;
+
+	try {
+		config = readConfig(
+			undefined,
+			{ ...args, experimentalJsonConfig: false, env },
+			true
+		);
+	} catch (err) {
+		if (
+			!(
+				err instanceof FatalError && err.code === EXIT_CODE_INVALID_PAGES_CONFIG
+			)
+		) {
+			throw err;
+		}
+	}
 
+	const nodejsCompat =
+		config?.compatibility_flags?.includes("nodejs_compat") ??
+		deploymentConfig.compatibility_flags?.includes("nodejs_compat") ??
+		false;
 	const defineNavigatorUserAgent = isNavigatorDefined(
-		deploymentConfig.compatibility_date,
-		deploymentConfig.compatibility_flags
+		config?.compatibility_date ?? deploymentConfig.compatibility_date,
+		config?.compatibility_flags ?? deploymentConfig.compatibility_flags
```

Comment:
I don't think this is correct. If the config is defined, it should _only_ check the config, and only otherwise check the deploymentConfig from the API.

Reply:
thank you for catching that! fixed